src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -27,508 +27,303 @@ public class ResultTextMarker
+                !SdkUIUtilities.TryGetWpfTextView(vsTextView, out IWpfTextView wpfTextView) ||
+                 vsTextView.GetBuffer(out IVsTextLines vsTextLines) != VSConstants.S_OK ||
+                 vsTextLines.GetLastLineIndex(out int lastLine, out int lastIndex) != VSConstants.S_OK)
             {
-                // Editor throws InvalidOperationException in some cases - 
-                // We act like tracking isn't turned on if this is thrown to avoid
-                // taking all of VS down.
+                return false;
             }
-        }
 
-        private bool IsTracking()
-        {
-            return _docCookie.HasValue && IsTracking(_docCookie.Value);
-        }
-
-        private void SubscribeToCaretEvents(IWpfTextView textView)
-        {
-            if (textView != null)
+            // If the start and end indexes are outside the scope of the text, skip tagging.
+            if (textSpan.iStartLine > lastLine ||
+                textSpan.iEndLine > lastLine)
             {
-                textView.Caret.PositionChanged += CaretPositionChanged;
-                textView.LayoutChanged += ViewLayoutChanged;
+                return false;
             }
-        }
 
-        private void ViewLayoutChanged(object sender, TextViewLayoutChangedEventArgs e)
-        {
-            if (_textView != null)
-            {
-                // If a new snapshot wasn't generated, then skip this layout
-                if (e.NewViewState.EditSnapshot != e.OldViewState.EditSnapshot)
-                {
-                    UpdateAtCaretPosition(_textView.Caret.Position);
-                }
-            }
-        }
+            // Move the end line to the start line if for some
+            // reason the end line is less than the start line.
+            textSpan.iEndLine = Math.Max(textSpan.iEndLine, textSpan.iStartLine);
 
-        private void CaretPositionChanged(object sender, CaretPositionChangedEventArgs e)
-        {
-            UpdateAtCaretPosition(e.NewPosition);
-        }
+            // Now fix up the column numbers.
+            ITextSnapshot textSnapshot = wpfTextView.TextSnapshot;
+            ITextSnapshotLine startTextLine = textSnapshot.GetLineFromLineNumber(textSpan.iStartLine);
+            ITextSnapshotLine endTextLine = textSnapshot.GetLineFromLineNumber(textSpan.iEndLine);
 
-        private void UpdateAtCaretPosition(CaretPosition caretPoisition)
-        {
-            // Check if the current caret position is within our region. If it is, raise the RegionSelected event.
-            if (_trackingSpan.GetSpan(_trackingSpan.TextBuffer.CurrentSnapshot).Contains(caretPoisition.Point.GetPoint(_trackingSpan.TextBuffer, PositionAffinity.Predecessor).Value))
+            // If the start column of the start lines is beyond the length of the start line
+            // then we will reset the start column to zero and maybe tag the entire line if the start and end lines are the same.
```

Comment:
>same [](start = 124, length = 4)

This _should_ violate the SARIF spec -- it should not be legal for `startColumn` to point past the end of the line -- but unfortunately the spec doesn't come right out and say that. I filed [oasis-tcs/sarif-spec#475](https://github.com/oasis-tcs/sarif-spec/issues/475), "Spec doesn't say that region.startColumn can't point past the end of the line".

Having said all that, we should consider treating this as if the spec got it right, and just returning false here.